cvFit/src/app/azureprofile/azureprofile.component.ts

```diff
@@ -0,0 +1,34 @@
+import { Component, OnInit } from '@angular/core';
+import { HttpClient } from '@angular/common/http';
+
+const GRAPH_ENDPOINT = 'https://graph.microsoft.com/v1.0/me';
+
+type ProfileType = {
+  givenName?: string,
+  surname?: string,
+  userPrincipalName?: string,
+  id?: string
+};
+@Component({
+  selector: 'app-azureprofile',
+  templateUrl: './azureprofile.component.html',
+  styleUrls: ['./azureprofile.component.css']
+})
+export class AZUREprofileComponent implements OnInit {
+  profile!: ProfileType;
+
+  constructor(
+    private http: HttpClient
+  ) { }
+
+  ngOnInit() {
+    this.getProfile();
+  }
+
+  getProfile() {
+    this.http.get(GRAPH_ENDPOINT)
+      .subscribe(profile => {
```

Comment:
You'll need to unsubscribe from this or set the observable to the profile variable and use `| async` in the view to let Angular handle the unsubscribe.